Add ws-federation docs (#5577)
wso2/docs-is: en/asgardeo/docs/get-started/try-samples/ws-federation-webapp.md

# Sample WS-Federation web app

By following this guide, you will be able to deploy a WS-Federation-based web application and enable login for it using the Passive Security Token Service (Passive STS) of {{product_name}}.

!!! info
    {{product_name}} uses its passive security token service (Passive STS) as its WS-Federation implementation.

>     Passive STS is capable of issuing SAML 1.1 and 2.0 security tokens. To request a SAML 2.0 security token, the Request Security Token (RST) should be sent to the passive STS endpoint with the token type, `SAMLV2.0`. If no RST is specified, {{product_name}} issued a SAML 1.1 token by default.

Reviewer: ```suggestion
    Passive STS is capable of issuing SAML 1.1 and 2.0 security tokens. To request a SAML 2.0 security token, the Request Security Token (RST) should be sent to the passive STS endpoint with the token type, `SAMLV2.0`. If no RST is specified, {{product_name}} issues a SAML 1.1 token by default.
```

Author: All the content in this PR is copied directly from the IS docs. @himeshsiriwardana, do we need to update the content?

Reviewer: If we have a corresponding IS doc with the same content, let's move that to /includes directory and reuse it for asgardeo using the {% include ""} directive.

Author: done

---

### Prerequisites

- Download [Apache Tomcat 8.x](https://tomcat.apache.org/download-80.cgi){:target="_blank"} and install it. Tomcat server installation location will later be referred to as `<TOMCAT_HOME>` in this guide.

- It is recommended that you use a hostname that is not `localhost` to avoid browser errors. Modify your machine's `/etc/hosts` entry to reflect this.

- Download the [Passive STS Sample application](https://github.com/wso2/samples-is/releases/download/v4.6.2/PassiveSTSSampleApp.war){:target="_blank"} from the latest release assets.

### Deploy the sample app

To deploy the sample web app on a web container:

1. Copy the application's downloaded `.war` file into the `webapps` directory of the Tomcat folder.

2. Start the Tomcat server.

> ### Configure sample properties

Reviewer: ```suggestion
### Configure application properties of the sample app
```

To configure additional properties for the sample application:

1. Add the following configurations to the `web.xml` file in `<TOMCAT_HOME>/apache-tomcat-<version>/webapps/PassiveSTSSampleApp/WEB-INF`.

>     - Specify `idpUrl` as the Identity Server's Passive STS URL.

Reviewer: ```suggestion
    - Specify Asgardeo Passive STS URL as `idpUrl`.
```

        ``` xml
        <init-param>
                <param-name>idpUrl</param-name>
                <param-value>https://api.asgardeo.io/t/<organization_name>/passivests</param-value>
        </init-param> 
        ```

>     - Specify the `replyURL` as the URL of the web app.

Reviewer: ```suggestion
    - Specify the URL of the web app as `replyURL`.
```

        ``` xml
        <init-param>
                <param-name>replyUrl</param-name>
                <param-value>http://localhost:8080/PassiveSTSSampleApp/index.jsp</param-value>
        </init-param>
        ```

>     - Specify the ` realm ` as a unique identifier for the web app.

Reviewer: ```suggestion
    - Specify a unique identifier for the web app as the `realm`.
```

        ``` xml
        <init-param>
                <param-name>realm</param-name>
                <param-value>PassiveSTSSampleApp</param-value>
        </init-param> 
        ```

2. Restart the tomcat server.

## Configure the service provider

> 1. On the WSO2 Identity Server Console, go to **Applications**.

Author: fixed

2. Click **New Application** and select **Standard-Based Application**.

3. Enter the following details:

    ![Create a new Passice STS app]({{base_path}}/assets/img/guides/applications/create-new-ws-federation-app.png){: width="700" style="display: block; margin: 0; border: 0.3px solid lightgrey;"}

    <table>
        <tr>
            <td>Name</td>
            <td>
                Give a unique name to identify your application.
                <p>e.g.:<code>PassiveSTSSampleApp</code></p>
            </td>
        </tr>
        <tr>
            <td>Protocol</td>

>             <td>Select <b>WS-Federation</b>.</td>

Reviewer: ```suggestion
            <td><b>WS-Federation</b>.</td>
```

    <table>
        <tr>
            <td>Reply Logout URL</td>
            <td>
                This endpoint in your application handles the logout response from {{product_name}}.
                <p><code>http://localhost:8080/PassiveSTSSampleApp/index.jsp</code></p>
            </td>
        </tr>
    </table>

6. Go to the **User Attributes** tab and click **Add User Attribute**, and add the following attributes:

    - `http://wso2.org/claims/username`
    - `http://wso2.org/claims/emailaddress`

7. Select `http://wso2.org/claims/emailaddress` as the **Subject attribute**.

8. Click **Update** to save your configurations.

## Try it out

!!! info
    When redirecting your users to {{product_name}} Passive STS endpoint, the following (optional) parameters are sent in the request from the sample application.

	- **wa=wsignin1.0**: specifies whether {{product_name}} should issue a token for the relying party (this is the default action).
    - **wa=wsignout1.0**: specifies whether {{product_name}} should log the user out.
    - **wreply={replyUrl}**: specifies where the response should be sent.

> 	Using a Network tracer such as a SAML tracer is recommended to analyze the HTTP request and responses in this scenario. With a tracer, you will be able to view the parameters mentioned above and also see the SAML token that is issued from {{product_name}}.

Reviewer: ```suggestion
	Use a Network tracer such as SAML tracer to analyze the HTTP request and responses in this scenario. With a tracer, you will be able to view the parameters mentioned above and also see the SAML token that is issued from {{product_name}}.
```

Maybe we can mark this as a "note" since this is additional information.
